keycloak_add_users.py

```diff
@@ -0,0 +1,209 @@
+        self.temporary = temporary
+        self.type = type
+        self.value = value
+
+    def toJson(self):
+        return [{'value': self.value, 'type': self.type, 'temporary': self.temporary}]
+
+
+class KeyCloakClient:
+    def __init__(self, baseUrl, username, password, accesstoken):
+        self.BaseURL = baseUrl
+        self.Username = username
+        self.Password = password
+        self.AccessToken = accesstoken
+
+    def startClient(self):
+        headers = {"Content-type": "application/x-www-form-urlencoded"}
+        response = self.rawMethod("POST", "/auth/realms/master/protocol/openid-connect/token",
+                        "grant_type=password&client_id=admin-cli&username={}&password={}".format(self.Username, self.Password), headers)
+        json_response = response.json()
+        self.AccessToken = json_response['access_token']
+
+    def Get(self, url, body, headers):
+        headers["Authorization"] = "Bearer {}".format(self.AccessToken)
+        return self.rawMethod("GET", url, body, headers)
+
+    def Post(self, url, body, headers):
+        headers["Authorization"] = "Bearer {}".format(self.AccessToken)
+        return self.rawMethod("POST", url, body, headers)
+
+    def Put(self, url, body, headers):
+        headers["Authorization"] = "Bearer {}".format(self.AccessToken)
+        return self.rawMethod("PUT", url, body, headers)
+
+    def putUser(self, realm_name, user):
+        headers = {"Content-type": "application/json"}
+        body = user.toJson()
+        response = self.Put("/auth/admin/realms/{}/users/{}".format(realm_name, user.id), body, headers)
+
+    def doesUserExist(self, realm_name, requestedUser):
+        return_dict = dict()
+        headers = {}
+        response = self.Get("/auth/admin/realms/{}/users/".format(realm_name), "", headers)
+        for user in response.json():
+            if user['username'] == requestedUser.username:
+                return_dict['user'] = user
+                return_dict['bool'] = True
+                return return_dict
+
+        return_dict['user'] = requestedUser
+        return_dict['id'] = 0
+        return_dict['bool'] = False
+        return return_dict
+
+    def rawMethod(self, method, url, body, headers):
+        fullUrl = self.BaseURL+url
+        try:
+            if (method == "GET"):
+                r = requests.get(fullUrl, headers=headers)
+            elif (method == "POST"):
+                r = requests.post(fullUrl, data=body, headers=headers)
+            elif (method == "PUT"):
+                r = requests.put(fullUrl, data=body, headers=headers)
+            else:
+                print("Not supported at this time")
+                exit(1)
+        except requests.exceptions.RequestException as e:
+            print("Error in request")
+            raise SystemExit(e)
+        return r
+
+    def addUser(self, realm_name, user):
+        rtn_dict = self.doesUserExist(realm_name, user)
+        if not rtn_dict['bool']:
```

Comment:
This can be simplified

```python
        if not rtn_dict['bool']:
            headers = {"Content-type": "application/json"}
            body = user.toJson()
            response = self.Post("/auth/admin/realms/{}/users".format(realm_name), body, headers)
        return rtn_dict['user']
```